drivers: udc: add bMaxPacketSize0 supported by the controller capability

jfischer-no opened the pull request:
drivers: udc: add bMaxPacketSize0 supported by the controller capability

jfischer-no committed:
drivers: udc: add capability for MPS of control endpoint

New capability is to be used by the stack to adjust the corresponding
field in the device descriptor.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
usb: device_next: update bMaxPacketSize0 based on controller capability

Update bMaxPacketSize0 field of the device descriptor based on
controller capability.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
usb: device_next: trigger dequeue of control IN endpoint on reset event

There might be pending data stage transfer not finished
while host performs a bus reset.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

desowin commented:
First 3 commits look fine (except for the repetition in commit message) and I think should go in as its own PR (this one). The DRAFT commit should probably be moved outside this PR.

jfischer-no committed:
drivers: udc: remove unused and unsupported udc_ep_flush()

udc_ep_flush() is not implemented by the driver and is not
used by the USB device stack. Remove it for less confusion
when porting drivers.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>
